lib/nested_filter.ex

```diff
@@ -5,57 +5,45 @@ defmodule NestedFilter do
 
   @doc """
   Take a (nested) map and filter out any keys with specified values in the
-  filter_values list.
+  values_to_reject list.
   """
-  @spec drop_by_value(map :: map(), filter_values :: list()) :: map()
-  def drop_by_value(map, filter_values) do
-    cond do
-      is_nested_map?(map) ->
-        new_map = map
-                  |> Enum.reduce(%{}, fn({key, val}, acc) ->
-                     Map.put(acc, key, drop_by_value(val, filter_values)) end)
-        Map.drop(new_map, filterable_keys(new_map, filter_values))
```

Comment:
This pipeline is doing a _lot_ of extra work; you're building the list of filterable keys each time though the reduce, then using that to inform the drop (which strangely, isn't in the pipeline, even though it could be). It's much cleaner and faster do do that work inside the reduce. If a key should be rejected, just return the accumulator unmodified.